add support for providing custom prooph persistence strategy

CipherdevNL opened the pull request:
Implementation for the ticket on the main ecotone repo:
https://github.com/ecotoneframework/ecotone/issues/14

Hopefully these changes are good enough, I'm not sure if the test coverage is as desired.
Wasn't able to test the changes locally with the automation tools, those result into the following error:

```
1) Test\Ecotone\EventSourcing\Integration\EventSourcingRepositoryBuilderTest::test_storing_and_retrieving
Error: Call to undefined method Ecotone\Enqueue\OutboundMessageConverter::unsetEnqueueMetadata()

/data/app/src/EventSourcing/EventSourcingRepository.php:84
/data/app/tests/EventSourcing/Integration/EventSourcingRepositoryBuilderTest.php:52
```

dgafka commented:
Hey @CipherdevNL 
Thanks for contribution :) 

The error you're having is related to outdated dependencies. 
You can rebase with master and run `composer update` locally. 

----

There is one error with behat test:

```
Test\Ecotone\EventSourcing\Behat\Bootstrap\DomainContext::iCreateBasketWithId()
      Some of the aggregate IDs or event IDs have already been used in the same stream. The PDO error should contain more information:
      Error 23505.
      Error-Info: ERROR:  duplicate key value violates unique constraint "_4117049273c6af1631d4c2ae34fb58fe54e2f765_pkey"
      DETAIL:  Key (no)=(1) already exists. (Prooph\EventStore\Exception\ConcurrencyException)
```

I will be happy to merge it after fixing the error. :) 
If you will be having difficulties with debugging it let me know and I will check it tomorrow / friday.

CipherdevNL commented:
Changed the code and resolved the broken test. Not sure why it wants to create a second event with the same event no. Changed the strategy to a different one and that one passes, so at least the code for setting custom strategies works.

Thanks for suggesting the `composer update`, already tried to rebase and reinstall composer, but didn't noticed that the `composer.lock` wasn't part of git. 😅

dgafka commented:
Thank you @CipherdevNL for contribution :) 
This is now available starting from `Ecotone 1.17.0`.
Docs are [updated](https://docs.ecotone.tech/modelling/event-sourcing#custom-strategy).
